File: javatesting/src/main/java/net/sahet/javatesting/Calc.java
package net.sahet.javatesting;

public class Calc {

	private static int x = 8;

	public int multiply(int a, int b) {
		System.out.println(x);
		if (a > b) {
			System.out.println("ok");
		} else {
			System.out.println("ko");
			x = 19;
			System.out.println(x);
		}
		return a * b;
	}

	public int foo(int x, int y) {
		int z = 0;
		if ((x > 0) && (y > 0)) {
			z = x;
		}
		return z;
	}

	/**
	 * Statement Test Coverage example
	 * 
	 * @param a
	 * @param b
	 */
	public int displaySum(int a, int b) {
		int sum = a + b;
		if (sum > 0) {
			System.out.println("Psotive sum = " + sum);
		} else if (sum < 0) {
			System.out.println("Negative sum = " + sum);
		}
//		else {
//			System.out.println("Zero");
//		}
		return sum;
	}

	public int calcSum(int x, int y) {
		if ((x == 0) || (y < 0)) {
			y = y / x; 
		} else {
			x = y + 2;
		}
		return x + y;
	}
}

File: fe-tests/jest-tests/src/calc.ts
export default class calc {
  static factorial(n: number): number {
    if (n === 0 || n === 1) return 1;
    return n * this.factorial(n - 1);
  }

  static fibonacci(n: number): number {
    if (n === 0 || n === 1) return 1;
    return this.fibonacci(n - 2) + this.fibonacci(n - 1);
  }

  static foo(x: number, y: number): number {
    let z: number = 0;
    if (x > 0 && y > 0) {
      z = x;
    }
    return z;
  }

  static displaySum(a: number, b: number): number {
    let sum = a + b;
    if (sum > 0) {
      console.log("Psotive sum = " + sum);
    } else if (sum < 0) {
      console.log("Negative sum = " + sum);
    }
    //		else {
    //			console.log(("Zero");
    //		}
    return sum;
  }

  static calcSum(x: number, y: number): number {
    if (x == 0 || y < 0) {
      y = y / x;
    } else {
      x = y + 2;
    }
    return x + y;
  }
}
